File: Java/1302_Deepest_Leaves_Sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        int maxDepth = getMaxDepth(root, 1);
        
        int sum = getNDepthSum(root, maxDepth, 1, 0);
        
        return sum;
    }
    public int getMaxDepth(TreeNode root, int depth){
        if(root.left == null && root.right == null){
            return depth;
        }
        else if(root.left == null){
            return getMaxDepth(root.right, depth+1);
        }
        else if(root.right == null){
            return getMaxDepth(root.left, depth+1);
        }
        else{
            return Math.max(getMaxDepth(root.left, depth+1), getMaxDepth(root.right, depth+1));
        }
    }
    public int getNDepthSum(TreeNode root, int n, int depth, int sum){
        if(n == depth){
            return sum + root.val;
        }
        else{
            if(root.left != null && root.right != null){
                return getNDepthSum(root.left, n, depth+1, sum) + getNDepthSum(root.right, n, depth+1, sum);
            }
            else if(root.left != null){
                return getNDepthSum(root.left, n, depth+1, sum);
            }
            else if(root.right != null){
                return getNDepthSum(root.right, n, depth+1, sum);
            }
            else{
                return 0;
            }
        }
    }
}

File: CS/1302_Deepest_Leaves_Sum.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int DeepestLeavesSum(TreeNode root) {
        int n = getN(root, 0);
        int sum = getSum(root, 1, n);
        return sum;
    }
    public int getSum(TreeNode root, int c, int n){
        if(root == null){
            return 0;
        }
        else if(c == n){
            return root.val;
        }
        else{
            return getSum(root.left, c+1, n) + getSum(root.right, c+1, n);
        }
    }
    public int getN(TreeNode root, int n){
        if(root == null){
            return n;
        }
        else{
            return Math.Max(getN(root.left, n+1), getN(root.right, n+1));
        }
    }
}
